src/synthesizer/csharp/mod.rs

```diff
@@ -0,0 +1,668 @@
+                ResourceIr::Array(_, array) => {
+                    if *idx <= array.len() {
+                        array[*idx].emit_csharp(output, root_resource);
+                    } else {
+                        output.text("null")
+                    }
+                }
+                other => {
+                    output.text(format!("Fn.Select({idx}, "));
+                    other.emit_csharp(output, root_resource);
+                    output.text(")")
+                }
+            },
+            ResourceIr::Cidr(cidr_block, count, mask) => {
+                output.text("Fn.Cidr(");
+                cidr_block.emit_csharp(output, root_resource);
+                output.text(", ");
+                count.emit_csharp(output, root_resource);
+                output.text(", ");
+                match mask.as_ref() {
+                    ResourceIr::Number(mask) => {
+                        output.text(format!("\"{mask}\""));
+                    }
+                    ResourceIr::String(mask) => {
+                        output.text(mask.to_string());
+                    }
+                    mask => mask.emit_csharp(output, root_resource),
+                }
+                output.text(")");
+            }
+        }
+    }
+}
+
+impl CsharpEmitter for OutputInstruction {
+    fn emit_csharp(&self, output: &CodeBuffer) {
+        let var_name = &self.name;
+
+        if let Some(cond) = &self.condition {
+            output.line(format!("{var_name} = {}", camel_case(cond)));
+            output.text(format!("{INDENT}? "));
+            let indented = output.indent(INDENT);
+            self.value.emit_csharp(&indented, None);
+            output.line(format!("\n{INDENT}: null;"));
+        } else {
+            output.text(format!("{var_name} = "));
+            self.value.emit_csharp(output, None);
+            output.line(";")
+        }
+
+        // Create CfnOutputs if the output is an export
+        if let Some(export) = &self.export {
+            if let Some(cond) = &self.condition {
+                let indented = output.indent_with_options(IndentOptions {
+                    indent: INDENT,
+                    leading: Some(format!("if ({}) {{", camel_case(cond)).into()),
+                    trailing: Some("}".into()),
+                    trailing_newline: true,
+                });
+                self.emit_cfn_output(&indented, export, var_name);
+            } else {
+                self.emit_cfn_output(output, export, var_name);
+            }
+        }
+    }
+}
+
+impl OutputInstruction {
+    fn emit_cfn_output(&self, output: &CodeBuffer, export: &ResourceIr, var_name: &str) {
+        let output = output.indent_with_options(IndentOptions {
+            indent: INDENT,
+            leading: Some(
+                format!(
+                    "new CfnOutput(this, \"{}\", new CfnOutputProps {{",
+                    &self.name
+                )
```

Comment:
I forget, does the name there == the output name in CFN?

Reply:
outputs don't have a name field. are you thinking of the export name?

This becomes the logical id for the output. 
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html